cmake/CodeCoverage.cmake

```diff
@@ -0,0 +1,427 @@
+# Copyright (c) 2012 - 2017, Lars Bilke
```

Comment:
I take it there are no local modifications to this, and it's just a verbatim import?

Reply:
I made some modifications, primarily adding `llvm-cov` support, and preventing failures if a non-Apple LLVM-based compiler is used (the original fails if GNU or Apple Clang is *not* used). I also removed the `lcov` function that was originally defined, since I didn't want to handle both `gcovr` and `lcov`. I'll add these points to the changelog comments in that file